### DISCUSSION TITLE
Reusable workflow input gets dropped

### DISCUSSION THREAD

**Discussion by jardon** [community/community#57836] [Actions] [answered]
### Select Topic Area

Question

### Body

I have built a reusable workflow for building artifacts for debian packages.  The workflows at least partially work, but I am running into some issues passing inputs.  Can someone explain why an input is seen in the job set up, but when referenced, it is blank?  You can see an example run [here](https://github.com/jardon/deb-ci/actions/runs/5245036861/jobs/9471948671).  Calling the workflow directly with a `workflow_dispatch` works flawlessly, but on a `workflow_call` the data seems to be fumbled somewhere.

Thanks in advance

**Comment by airtower-luna** [comment, marked as answer]
The reason is described in the [`workflow_call` event documentation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_call):

> When a workflow is triggered with the `workflow_call` event, the event payload in the called workflow is the same event payload from the calling workflow.

So the arguments in the `with` section simply don't exist under `github.event.inputs`. You should use the [`inputs` context](https://docs.github.com/en/actions/learn-github-actions/contexts#inputs-context) instead.
